### PR TITLE
Hide ProblemBuilder.build

### PR DESCRIPTION
**Pull request by reinsch82**
private #3965

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/catalog/DefaultVersionCatalogBuilder.java [hunk 1]
```diff
@@ -225,6 +217,15 @@ private DefaultVersionCatalog doBuild() {
         return new DefaultVersionCatalog(name, description.getOrElse(""), realizedLibs.build(), ImmutableMap.copyOf(bundles), ImmutableMap.copyOf(versionConstraints), realizedPlugins.build());
     }
 
+    private static ProblemBuilder configureVersionCatalogError(ProblemBuilderDefiningLabel builder, String message, VersionCatalogProblemId catalogProblemId) {
```
Comment (octylFractal) [inline subprojects/dependency-management/src/main/java/org/gradle/api/internal/catalog/DefaultVersionCatalogBuilder.java:220]: 💅 Can we keep this next to `throwVersionCatalogProblemException`?

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/catalog/DefaultVersionCatalogBuilder.java [hunk 2]
```diff
@@ -177,40 +178,31 @@ public void withContext(String context, Runnable action) {
...
-    @Nonnull
-    public ProblemBuilder createVersionCatalogError(String message, VersionCatalogProblemId catalogProblemId) {
-        return getProblemService().createProblemBuilder()
-            .label(message)
-            .documentedAt(userManual(VERSION_CATALOG_PROBLEMS, catalogProblemId.name().toLowerCase()))
-            .noLocation()
-            .type(catalogProblemId.name())
-            .group(VERSION_CATALOG_ID)
-            .severity(ERROR);
+    private RuntimeException throwVersionCatalogProblemException(ReportableProblem problem) {
```
Comment (octylFractal) [inline subprojects/dependency-management/src/main/java/org/gradle/api/internal/catalog/DefaultVersionCatalogBuilder.java:181]: 💭 If `createVersionCatalogError` was made `static`, perhaps this should also be made `static`?

**Comment by reinsch82** [comment]
@bot-gradle test this

**Review by octylFractal — APPROVED** [review]
LGTM for the JVM platform changes.

**Comment by reinsch82** [comment]
@bot-gradle test this

**Review by donat — APPROVED** [review]

**Comment by reinsch82** [comment]
@bot-gradle test and merge

**Pull request merged**